lib/ansible/modules/windows/win_scheduled_task.ps1

```diff
@@ -135,19 +152,25 @@ try {
         if ($frequency -eq "once") {
             $trigger = New-ScheduledTaskTrigger -Once -At $time
         }
+        elseif ( $frequency -eq "interval"){
+            $trigger = New-ScheduledTaskTrigger -Once -RepetitionInterval (iex $("new-timespan -$interval_unit $interval")) -At $time -RepetitionDuration ([timespan]::MaxValue)
+        }
+        elseif ( $frequency -eq "hourly"){
+            $trigger = New-ScheduledTaskTrigger -Once -RepetitionInterval (new-timespan -hour 1) -At $time -RepetitionDuration ([timespan]::MaxValue)
+        }
         elseif ($frequency -eq "daily") {
             $trigger = New-ScheduledTaskTrigger -Daily -At $time
         }
         elseif ($frequency -eq "weekly"){
             $trigger = New-ScheduledTaskTrigger -Weekly -At $time -DaysOfWeek $days_of_week
         }
         else {
-            Fail-Json $result "frequency must be daily or weekly"
+            Fail-Json $result "frequency must be once, interval, hourly, daily or weekly"
         }
     }
 
     if ( ($state -eq "absent") -and ($exists) ) {
-        Unregister-ScheduledTask -TaskName $name -Confirm:$false -WhatIf:$check_mode
+        Unregister-ScheduledTask -TaskName $name -TaskPath $path -Confirm:$false -WhatIf:$check_mode
```

Comment:
We should put in the -TaskPath flag here to guarantee that we are removing the task from the desired folder.